crates/proof-of-sql/src/sql/postprocessing/group_by_postprocessing.rs

```diff
@@ -169,6 +177,153 @@ impl GroupByPostprocessing {
+                    .unzip()
+            })
+            .unwrap_or((vec![], vec![]));
+        let count_ids: Vec<_> = evaluated_columns
+            .get(&AggregationOperator::Count)
+            .unwrap_or(&vec![])
+            .iter()
+            .map(|(id, _)| *id)
+            .collect::<Vec<_>>();
+        let aggregation_results = aggregate_columns(
+            &alloc,
+            &group_by_ins,
+            &sum_ins,
+            &max_ins,
+            &min_ins,
+            &selection_in,
+        )?;
+        // Finally do another round of evaluation to get the final result
+        // Gather the results into a new OwnedTable
+        let group_by_outs: IndexMap<Identifier, OwnedColumn<S>> = aggregation_results
+            .group_by_columns
+            .iter()
+            .zip(self.group_by_identifiers.iter())
+            .map(|(column, id)| (*id, OwnedColumn::from(column)))
+            .collect::<IndexMap<_, _>>();
+        let sum_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.sum_columns.iter(),
+            sum_ids.iter(),
+            sum_ins.iter(),
+        ))
+        .map(|(c_out, id, c_in)| {
+            Ok((
+                *id,
+                OwnedColumn::try_from_scalars(c_out, c_in.column_type())?,
+            ))
+        })
+        .collect::<PostprocessingResult<_>>()?;
+        let max_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.max_columns.iter(),
+            max_ids.iter(),
+            max_ins.iter(),
+        ))
+        .map(|(c_out, id, c_in)| {
+            Ok((
+                *id,
+                OwnedColumn::try_from_option_scalars(c_out, c_in.column_type())?,
+            ))
+        })
+        .collect::<PostprocessingResult<_>>()?;
+        let min_outs: IndexMap<Identifier, OwnedColumn<S>> = multizip((
+            aggregation_results.min_columns.iter(),
+            min_ids.iter(),
+            min_ins.iter(),
+        ))
+        .map(|(c_out, id, c_in)| {
+            Ok((
+                *id,
+                OwnedColumn::try_from_option_scalars(c_out, c_in.column_type())?,
+            ))
+        })
+        .collect::<PostprocessingResult<_>>()?;
+        //TODO: When we have NULLs we need to differentiate between count(1) and count(expression)
+        let count_column = OwnedColumn::BigInt(aggregation_results.count_column.to_vec());
+        let count_outs: IndexMap<Identifier, OwnedColumn<S>> = count_ids
+            .iter()
+            .map(|id| (*id, count_column.clone()))
+            .collect::<IndexMap<_, _>>();
+        let new_owned_table = OwnedTable::try_from_iter(
+            group_by_outs
+                .iter()
+                .chain(sum_outs.iter())
+                .chain(max_outs.iter())
+                .chain(min_outs.iter())
+                .chain(count_outs.iter())
+                .map(|(id, column)| (*id, column.clone())),
+        )?;
```

Comment:
See other ⭐ comments. I think this could end up looking like this:
```suggestion
        let new_owned_table = group_by_outs
            .chain(sum_outs)
            .chain(max_outs)
            .chain(min_outs)
            .chain(count_outs)
            .process_results(|iter| OwnedTable::try_from_iter(iter))??;
```